packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/heatmap.dart

```diff
@@ -0,0 +1,357 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart'
+    show immutable, listEquals, objectRuntimeType;
+import 'package:flutter/material.dart' show Color;
+
+import 'types.dart';
+
+/// Uniquely identifies a [Heatmap] among [GoogleMap] heatmaps.
+///
+/// This does not have to be globally unique, only unique among the list.
+@immutable
+class HeatmapId extends MapsObjectId<Heatmap> {
+  /// Creates an immutable identifier for a [Heatmap].
+  const HeatmapId(super.value);
+}
+
+/// A wrapper around platform specific behavior for the radius of a [Heatmap].
+///
+/// Currently this class does nothing. See https://github.com/flutter/flutter/issues/145411
+@immutable
+class HeatmapRadius {
+  const HeatmapRadius._(this.radius);
+
+  /// Create a [HeatmapRadius] with a radius in pixels.
+  ///
+  /// This value will be used verbatim on all platforms. It is up to the
+  /// developer to ensure that the value is appropriate for the platform.
+  const HeatmapRadius.fromPlatformSpecificValue(int radius) : this._(radius);
+
+  /// Create a [HeatmapRadius] from a platform value.
+  ///
+  /// In the future, this will do the opposite conversion that [pixels] does.
+  const HeatmapRadius.fromJson(dynamic json) : this._(json as int);
+
+  /// The platform-independent value of the radius.
```

Comment:
I refactored this class. Let me know if that helped clear things up.

Reply:
Is radius the equivalent to Density independent pixels or are there more platform calculations that need to take place?

Reply:
> Is radius the equivalent to Density independent pixels or are there more platform calculations that need to take place?

This is the thing I was telling you about offline; the answer unfortunately is that currently we don't know what it's equivalent to. It's hard to find the specific discussion to link to since it's a long PR at this point, but as far experimentation was able to determine, it's not just device pixel ratio.

I still don't think we want two different members here though. I think we just want `radius`, and it should be commented as something like `The radius that will be used by the underlying SDK`. We don't need to separately preserve the value passed into `fromPixels`.